### ISSUE TITLE
Getting unrecognized selector error in AWSURLResponseSerialization:responseObjectForResponse

### ISSUE THREAD

**Issue by croesus** [aws/aws-sdk-ios#343]
When downloading an HTML file from an S3 bucket, line 270 of AWSURLResponseSerialization.m generates the following error:
`-[NSURL bytes]: unrecognized selector sent to instance 0x7faf03866eb0`

The code that incorrectly passes an NSURL* object to a NSData* parameter (which doesn't cause a compile time error because the parameter is typed as 'id') is on line 310 of AWSURLSessionManager.m:

```objc
delegate.responseObject = [delegate.request.responseSerializer responseObjectForResponse:httpResponse
                                                                         originalRequest:sessionTask.originalRequest
                                                                          currentRequest:sessionTask.currentRequest
                                                                                    data:delegate.downloadingFileURL // <-- Incorrect type
                                                                                   error:&error];
```

A little lower down, the code at line 324 (executed when `delegate.shouldWriteToFile` is `NO`) passes `delegate.responseData` to this parameter, which is presumably the correct data to pass.

**Comment by croesus** [comment]
A little more detail. This error occurs whenever I try to download a file with content-type text/html from an S3 bucket using `AWSS3TransferManager download:`. If I set the content-type header manually for the HTML file to e.g. text/plain, the file is downloaded successfully. 

I've created a small single-view app using a test Cognito identity pool to illustrate the bug - [source](https://s3-eu-west-1.amazonaws.com/racefully-awstest-public/S3Test.zip)

You'll need to run `pod install` in `/S3Test` to get the AWSS3 pod and then open the workspace to build the file.

**Comment by yosuke-matsuda** [comment]
There is logic in the SDK to categorize the response as an error when the returned content type is HTML. It seemed to be implemented to address some edge cases where the error response is not properly processed. It, however, becomes an issue when you get a valid HTML object as a response. We will take a look into this. Thanks.

**Comment by karthiksaligrama** [comment]
This issue should be fixed with 2.4.2
